src/panels/KaitoManagePanel.ts

```diff
@@ -0,0 +1,206 @@
+                writeFileSync(tempFilePath, modelYaml, "utf8");
+                const command = `apply -f ${tempFilePath}`;
+                const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+                if (failed(kubectlresult)) {
+                    vscode.window.showErrorMessage(`Error deploying 'workspace-${modelName}': ${kubectlresult.error}`);
+                    return;
+                }
+            });
+            vscode.window.showInformationMessage(`'workspace-${modelName}' has been redeployed.`);
+            await this.updateModels(webview);
+        } finally {
+            this.operating[modelName] = false;
+        }
+    }
+
+    // Updates the current state of models on the cluster
+    private async updateModels(webview: MessageSink<ToWebViewMsgDef>) {
+        const command = `get workspace -o json`;
+        const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        if (failed(kubectlresult)) {
+            webview.postMonitorUpdate({
+                clusterName: this.clusterName,
+                models: [],
+            });
+            return;
+        }
+
+        const models = [];
+        const data = JSON.parse(kubectlresult.result.stdout);
+        for (const item of data.items) {
+            const conditions: Array<{ type: string; status: string }> = item.status?.conditions || [];
+            const { resourceReady, inferenceReady, workspaceReady } = getConditions(conditions);
+            models.push({
+                name: item.inference.preset.name,
+                instance: item.resource.instanceType,
+                resourceReady: resourceReady,
+                inferenceReady: inferenceReady,
+                workspaceReady: workspaceReady,
+                age: convertAgeToMinutes(item.metadata?.creationTimestamp),
+            });
+        }
+
+        webview.postMonitorUpdate({
+            clusterName: this.clusterName,
+            models: models,
+        });
+    }
+
+    private async handleMonitorUpdateRequest(models: ModelState[], webview: MessageSink<ToWebViewMsgDef>) {
+        void models;
+        await this.updateModels(webview);
+    }
+
+    // Retrieves the logs from the Kaito workspace and outputs them in a new text editor.
+    private async handleGetLogsRequest() {
+        await longRunning(`Retrieving logs`, async () => {
+            let command = `get po -l app=ai-toolchain-operator -A -o json`;
+            let kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            if (failed(kubectlresult)) {
+                vscode.window.showErrorMessage(`Error fetching logs: ${kubectlresult.error}`);
+                return;
+            }
+            const data = JSON.parse(kubectlresult.result.stdout);
+            const items = data.items;
+            let pod = { name: null, date: new Date(`2000-01-01T12:00:00Z`) };
```

Comment:
Suggestion here: any idea why this date is used from year 2000? you can leave a comment about it. below the logic compares with pod creation time and it is going to be always greater than year 2000. instead you can just stamp the date as creationTimestamp. (comparison logic can be avoided)

Reply:
May be try the following avoiding placeholder date and find the pod with name "kaiot-workspace"

```
let pod = null;
for (const item of items) {
    const name = item.metadata.name;
    if (name.startsWith("kaito-workspace")) {
        const date = new Date(item.metadata.creationTimestamp);
        if (!pod || date > pod.date) {
            pod = { name: name, date: date };
        }
    }
}
```